### PR TITLE
Implements projection inside reactive MongoDB with Panache

### PR DESCRIPTION
**Pull request by loicmathieu**
@cescoffier this is a leftover from the reactive implementation of MongoDB with Panache.
I forget to implements the `project(Class)` methods that was merged recently.

### CODE REVIEW

**Comment by loicmathieu** [comment]
@cescoffier can you review it ? 
@gsmet It will be good to have it for 1.3 as it's the only difference between the imperative and the reactive implementation.

**Comment by loicmathieu** [comment]
@cescoffier can you review it ? It's a missing part in the reactive MongoDB with panache implementation.

File: extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/PanacheQueryImpl.java [hunk 1]
```diff
@@ -40,7 +39,6 @@
     PanacheQueryImpl(MongoCollection<? extends Entity> collection, Class<? extends Entity> entityClass, Document mongoQuery,
```
Comment (FroMage) [inline extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/PanacheQueryImpl.java]: Might as well remove the parameter if we're not using it, no?

**Review by FroMage — CHANGES_REQUESTED** [review]
Aside from the param to remove, LGTM.

File: extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/PanacheQueryImpl.java (hunk 1 above)
Comment (loicmathieu) [reply extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/PanacheQueryImpl.java]: Yes, this was copied from Hibernate with Panache but not needed here.
I remove it (and also for the reactive implementation), a little cleanup doesn't hurt ;)

**Review by FroMage — APPROVED** [review]

**Pull request merged**